components/ArticlePageLayout.js

```diff
@@ -227,7 +226,10 @@ function ArticlePageLayout({
     </mark>
   );
 
-  const queryString = querystring.stringify(query);
+  // if there's no repliedByMe in filter, we needn't userId field
+  if (!listQueryVars.articleRepliesFrom) delete queryWithOptions.userId;
+  const queryString = querystring.stringify(queryWithOptions);
```

Comment:
I was thinking if we can directly convert `listQueryVars` to JSON and put them in feed URL. That should simplify `api/articles/[feed]` (because it don't need `getQueryVars`, just `JSON.parse`) and allows greater flexibility for streams.

What do you think @nonumpa ?

Reply:
I also wanted to remove the dependency of `ArticlePageLayout.getQueryVars` from `api/articles/[feed]`, but it seems that it's not easy to directly convert `listQueryVars` to JSON.
![螢幕快照 2020-08-05 下午6 01 23](https://user-images.githubusercontent.com/6376572/89401059-5102a880-d747-11ea-8185-e96c7c51b9c3.png)
I'm afraid that I should write a `listQueryVarsToJSON` function and another `getQueryVars` for `api/articles/[feed]`.

Reply:
I was thinking a direct `JSON.stringify` should do the job :P
Is that too long for an URL?

Reply:
Sorry, I mean querystring, not JSON.
Maybe I can encode filter and orderBy like [this](
https://stackoverflow.com/questions/41641426/passing-objects-as-url-parameters-c-sharp).